lib/karafka/connection/delegator.rb

```diff
@@ -20,39 +20,39 @@ def call(group_id, kafka_messages)
           # @see topic_mapper internal docs
           mapped_topic_name = Karafka::App.config.topic_mapper.incoming(kafka_messages[0].topic)
           topic = Routing::Router.find("#{group_id}_#{mapped_topic_name}")
-          controller = Persistence::Controller.fetch(topic, kafka_messages[0].partition) do
-            topic.controller.new
+          consumer = Persistence::Consumer.fetch(topic, kafka_messages[0].partition) do
+            topic.consumer.new
           end
 
-          # Depending on a case (persisted or not) we might use new controller instance per each
+          # Depending on a case (persisted or not) we might use new consumer instance per each
```

Comment:
Depending on a case (persisted or not) we might use new consumer instance per each batch, or use the same one for all of them (for implementing buffering, etc.)